crates/matrix-sdk-crypto/src/olm/group_sessions/outbound.rs

```diff
@@ -429,12 +486,15 @@ impl OutboundGroupSession {
     pub(crate) fn is_shared_with(&self, device: &Device) -> ShareState {
         // Check if we shared the session.
         let shared_state = self.shared_with_set.get(device.user_id()).and_then(|d| {
-            d.get(device.device_id()).map(|s| {
-                if Some(s.sender_key) == device.curve25519_key() {
-                    ShareState::Shared(s.message_index)
-                } else {
-                    ShareState::SharedButChangedSenderKey
+            d.get(device.device_id()).map(|s| match s.value() {
+                ShareInfo::Shared(s) => {
+                    if Some(s.sender_key) == device.curve25519_key() {
```

Comment:
Just me or would the method call be less surprising on the left hand side of `==`?